source/slang/emit.cpp

```diff
@@ -6765,6 +6740,34 @@ String emitEntryPoint(
 #endif
         validateIRModuleIfEnabled(compileRequest, irModule);
 
+        // For GLSL only, we will need to perform "legalization" of
+        // the entry point and any entry-point parameters.
+        //
+        // TODO: We should consider moving this legalization work
```

Comment:
At this point I think we could remove the `TODO` comment here, since we've moved the pass much later, which was the original point.